### ISSUE TITLE
iTunes account blocked/corrupted by fastlane sigh download

### ISSUE THREAD

**Issue by elorz007** [fastlane/fastlane#10793]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

After using `fastlane sigh download` to manually download provisioning profiles, the itunes account that was used gets somehow blocked. When signing in to appleid.apple.com I get the error "**Failed to verify your identity. Try again**" and the page stays with the loading indicator forever. 

![screen shot 2017-11-03 at 18 36 32](https://user-images.githubusercontent.com/3601038/32388801-1bda896c-c0c9-11e7-95c5-45314b59091c.png)

With the same account I can log in to developer.apple.com but **not** to itunesconnect.apple.com

After some time or some action that I cannot pinpoint, the account works again (asks for the security questions every time you log in but it works). Then `fastlane sigh download` works a couple of times and right after that the accounts gets blocked again.

I tried with a complete new apple id, registered everything in the development portal, etc. Then I used `fastlane sigh download` couple of times, it worked and then the **new account also got blocked**. 

Needless to say, I tried resetting the password (successfully), different computers, etc. Didn't help.

When the account is in this blocked state, when using `fastlane sigh download` I get the following error:

```
[18:28:16]: ▸ [18:28:16]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[18:28:18]: ▸ +-------------------------------------+-------------------------------------------+
[18:28:18]: ▸ |                             Summary for sigh 2.63.0                             |
[18:28:18]: ▸ +-------------------------------------+-------------------------------------------+
[18:28:18]: ▸ | development                         | true                                      |
[18:28:18]: ▸ | provisioning_name                   | <Redacted>      
[18:28:18]: ▸ | username                            | <Redacted>                    
[18:28:18]: ▸ | app_identifier                      | <Redacted> 
[18:28:18]: ▸ | skip_certificate_verification       | true                                      |
[18:28:18]: ▸ | team_id                             | <Redacted>                                |
[18:28:18]: ▸ | adhoc                               | false                                     |
[18:28:18]: ▸ | skip_install                        | false                                     |
[18:28:18]: ▸ | force                               | false                                     |
[18:28:18]: ▸ | ignore_profiles_with_different_name | false                                     |
[18:28:18]: ▸ | skip_fetch_profiles                 | false                                     |
[18:28:18]: ▸ | platform                            | ios                                       |
[18:28:18]: ▸ | readonly                            | false                                     |
[18:28:18]: ▸ +-------------------------------------+-------------------------------------------+
[18:28:18]: ▸ [18:28:18]: Starting login with user '<Redacted>'
[18:28:20]: ▸ Looking for related GitHub issues on fastlane/fastlane...
[18:28:20]: ▸ 
[18:28:20]: ▸ [!] The request could not be completed because:
[18:28:20]: ▸ <html>
[18:28:20]: ▸ <head><title>503 Service Temporarily Unavailable</title></head>
[18:28:20]: ▸ <body bgcolor="white">
[18:28:20]: ▸ <center><h1>503 Service Temporarily Unavailable</h1></center>
[18:28:20]: ▸ <hr><center>Shield</center>
[18:28:20]: ▸ </body>
[18:28:20]: ▸ </html>
[18:28:20]: ▸ 
```

I would really appreciate any suggestions here even though it looks like a problem on apple's side, something very specific to my accounts or something temporary...

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.2                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.9.2                           |
| Installation Source         | ~/.rbenv/versions/2.2.2/bin/fastlane        |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.rbenv/versions/2.2.2/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.63.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.4        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.2        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.6       |
| faraday_middleware        | 0.10.1       |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| addressable               | 2.5.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
| rubyzip                   | 1.2.0        |
</details>

*generated on:* **2017-11-03**
</details>

**Comment by acerbetti** [comment]
Same happened to me

**Comment by bartoszj** [comment]
I've exactly the same problem :(

**Comment by elorz007** [comment]
Update: I still can't sign in but some of the `fastlane sigh download` complete with success.
And also I'm seeing the following line:
```
[09:05:10]: fastlane failed to write the crash report to /Users/ios/.fastlane/latest_crash.json.

**Comment by elorz007** [comment]
Weirdly enough, if I sign in to developer.apple.com and then navigate to itunesconnect.apple.com I stayed logged in. But if I sign in to developer.apple.com but then go appleid.apple.com then I'm automatically logged out of everything.

**Comment by ytn3rd** [comment]
Logging into appleid.apple.com the browser console log outputs "POST https://idmsa.apple.com/appleauth/auth/signin 503 (Service Temporarily Unavailable)" which is the same error that is output.

**Comment by bolom** [comment]
Same error but with Spaceship since 3 days too

**Comment by robbiet480** [comment]
Also experiencing this...

**Comment by casz** [comment]
Also experiencing this 😭

**Comment by casz** [comment]
@KrauseFx so this is an issue.
Signing into Developer portal and from there we can go to iTunesConnect. (After resetting password)

However when using Fastlane or logging in directly to iTunesConnect we get the infamous "Failed to verify your identity"

**Comment by taquitos** [comment]
thanks @casz, we're checking it out 👍

**Comment by taquitos** [comment]
If you have a proxy or VPN, we've found that you can get around this by enabling it, or disabling it. This might be some new rate limiting based on your location.

**Comment by bolom** [comment]
same as you @taquitos. or wait around 15mn

**Comment by AndyIsaacson** [comment]
@taquitos - interesting... we're building on CircleCI and hitting the problem a bunch.  That'd be pretty bad if all of Circle was in the same rate-limited pool!

**Comment by robbiet480** [comment]
I'm having this issue without a VPN or proxy FYI

**Comment by taquitos** [comment]
Hey @robbiet480, yeah, the problem isn't limited to VPN or Proxy, but using a VPN or Proxy is a workaround. That's because Apple seems to be rate limiting people based on IP.

**Comment by taquitos** [comment]
Ok, I've tested https://github.com/fastlane/fastlane/pull/10826 on my machine and it seems to be working.

**Comment by taquitos** [comment]
Ok, after some more testing of the problem, I've found the following Work arounds
- Login using a different account from same IP
- Change IP address (VPN/Proxy)
- Login from a different IP, it unblocks your original IP.
- Login in developer.apple.com and then navigate to itunesconnect.apple.com

So the IP blocking portion doesn't seem to apply to different accounts now (not sure if my tests yesterday were bad, or if Apple already made a small change)

**Comment by levibostian** [comment]
@taquitos Thank you. That info has helped me to get my release out today. 

Here is some more info if it is helpful. 

I have an app that I have white labeled. When I run Fastlane, I have 4 separate schemes that I run through `pem`, `produce`, `match`, and `register_devices` all at one time to make sure that everything is always in sync. When I want to make a beta release or a release to the App Store for production, I always run all 4 schemes through all of those tasks to make sure that everything is in sync. 

Before today, I have never had an issue with running all of these tasks but today I am running into the 503. 

This is how far I can get:

1. `produce` on app A. Sync with DevCenter and iTC.
2. `pem` on app A.
3. `produce` on app B. Sync with DevCenter and iTC.
4. `pem` on app B.
5. `produce` on app C. Sync with DevCenter *but not iTC*. That is where I am getting my 503. 
...If I try and run Fastlane again after I hit the 503 without changing my IP address, I instantly gt 503 instead of getting even to run `produce` on app A.

If I use VPN to change my IP address, I get the same result as above ^^^. I use VPN again to change my IP address again and I get the same result as above ^^^. This is consistent. 

So, what I did was comment out the `pem` track in my Fastfile to try and have Fastlane run 4 less tasks (running pem for al 4 separate schemes) and I am able to run Fastlane to release my app no problem. That makes me wonder if they are rate limiting by the number of tasks in a certain amount of time?

**Comment by taquitos** [comment]
Thanks for this info! This is super helpful to understand how people are running into this situation. We're going to start exploring more aggressive session caching so if you use the same account, we don't try to login each time. I think that should help a ton.

**Comment by casz** [comment]
@taquitos Yes PLEASE 💯 

I can give another example of how we use it.

We have an lane that updates provision profiles and certs using match for all our identifiers, that includes our app team, enteprise team and our white labeled teams. This however is all using the same account.

This lane even adds new devices to each team using different device pool for each as that is defined in our ruby code.

We set it up this way to make it easy for developers to sent PR to add new devices and identifiers :)
It can even produce newly arrived bundle identifiers 🙊 

Then use the same lane to ensure all our build agents have the same provision profiles and certs while using read only 👍

**Comment by taquitos** [comment]
@casz that makes a ton of sense. We definitely want to, and are working hard to enable setups like yours. This is such great usage information for us to know about. Thank you! ❤️ ❤️ ❤️

**Comment by taquitos** [comment]
I wanted to update folks around the auth issues we’re seeing right now.
Context:
Issue first seen in https://github.com/fastlane/fastlane/issues/10793
Picked up again in https://github.com/fastlane/fastlane/issues/10812

While I have a PR up that fixes this for the simple login case, it only addresses SMS 2FA, and it employs web scraping, which is super flakey. 

Furthermore, the authentication logic we already have addresses so many corner cases, and is battle tested. Replacing that with this flakey update could cause a lot of pain for everybody.

Instead, we think the right thing moving forward is to assume Apple isn’t going to fix this obvious issue, and then focus our energy on improving our session caching, so that between various actions, we don’t login multiple times unless explicitly called for.

This will take some time to properly address, but we’ll have a plan soon and we’ll share that out as soon as we do 🚀. In the meantime, we’re going to keep my PR (https://github.com/fastlane/fastlane/pull/10826) alive if it is helping to unblock people.

Repeating what I posted before, we have some workarounds whose success seem to vary:

- Login using a different account from same IP
- Change IP address (VPN/Proxy)
- Login from a different IP, it unblocks your original IP.
- Login in developer.apple.com and then navigate to itunesconnect.apple.com
-

**Comment by ytn3rd** [comment]
> Instead, we think the right thing moving forward is to assume Apple isn’t going to fix this obvious issue

Is it an Apple issue/bug, or is it Apple circumventing tools like fastlane?

**Comment by taquitos** [comment]
@ytn3rd one theory is that Apple is trying to prevent brute-force login attacks, but they aren't reenabling logins after a successful login. 

I say it's probably an Apple issue because you can get this to trigger even on the website. That being said, it's probably really rare for people not using automation like fastlane.

**Comment by dabvid** [comment]
I think were facing the same root cause when running PEM on multiple targets.  Our pem script runs fine for 8 targets then for the following were getting "503 Service Temporarily Unavailable."
Does anyone know if a 15 second sleep would do for us as a workaround until this issue is resolved?

**Comment by dabvid** [comment]
I can confirm a 15 second sleep between the logins through pem did work for 20+ targets. Not a real solution but might perhaps help someone.

**Comment by yfedin** [comment]
Having the same issue... 

Have 6 separate build jobs each executing fastlane cert/sigh/gym (for various combinations of target/configuration/backend). All are triggered by a single Tests job and are run across 2 nodes with 2 executors on each. 

Most of those (or at times - all of them) fail due to "503 Service Temporarily Unavailable." on cert step. So far didn't find any good workaround.

**Comment by yfedin** [comment]
And I can also confirm that code in branch [10793](/fastlane/fastlane/tree/10793) (PR #10826) does fix the issue for me. As I am not using 2FA (which turned to be a blocker to push this PR further) -
will stick with it for now. Is there any ETA for a proper fix for this issue?

**Comment by casz** [comment]
@taquitos @KrauseFx perhaps an survey, "tell us how you use fastlane" would make sense? 😝

**Comment by bartoszj** [comment]
In our company, we have reported this issue on https://developer.apple.com/contact/apple-id/. Few email later an "senior Advisor with the Apple Developer Program Support" wrote an odd message:

> thank you for letting us know that the issue is resolved and providing us with the additional information.

We didn't let them know that the issue was resolved :/, but since then we haven't received this error anymore. Maybe they increase some "limits", whitelist our IP address, or something else, but I suggest to contact Apple at https://developer.apple.com/contact/apple-id/.

**Comment by casz** [comment]
@bartoszj thanks for the link, there is so many ways to contact them and yet we got zero response :|
However the issue does seem to have fixed itself. Though I still suggest that more aggressive session caching should be added.

**Comment by elorz007** [comment]
We use sigh to download and install around 6 provisioning profiles. This is done on a jenkins job triggered after each update to any pull request. This job is done around 20 times per day and also 4 or 5 jobs can run in parallel (we have around 8 nodes). Meaning we do lots and lots of log ins in very short times and from different machines.

**Comment by elorz007** [comment]
I also contacted apple support some days ago. They said literally "Very weird, I have never seen this before" and immediately gave me the direct phone number of an apple support specialist. He didn't see any problem and didn't give any solution.

My guess is that some firewall/dos detecting hardware/network infrastructure changed on the Apple side and now it spits these 503 errors upon detecting certain patterns. I guess it is quite low-level stuff because the password is not even checked, the response is exactly the same with wrong or with right password. And that might be the reason why even the apple support doesn't know how to solve it.

**Comment by sirdharma** [comment]
https://github.com/fastlane/fastlane/pull/10826 fixes the issue for me. Thank you!

The following workarounds did not work for me:
> - Login using a different account from same IP
> - Login from a different IP, it unblocks your original IP.
> - Login in developer.apple.com and then navigate to itunesconnect.apple.com

I did not try the following:
> - Change IP address (VPN/Proxy)

**Comment by taquitos** [comment]
@sirdharma, that's great, we're still working on our session caching as our more permanent solution 🎈:goat:

**Comment by bartoszj** [comment]
After four days I got once again this error. Apple Support has responded with:

> Thank you again for your email. I understand that you are still getting the error 503 occasionally.
>
> The error message that you are being presented with is not an error that I am able to assist you with here at Apple Developer Program Support as this a server error that you are getting and we do not provide third party support. 
>
> I hope that this helps you. Thank you again and have a wonderful day. 

So they will not help us.

**Comment by taquitos** [comment]
@bartoszj you can trigger this error from the web too, so I wonder if you show them a screenshot of the login page not allowing you to login, that might change their minds about being able to help?

**Comment by bartoszj** [comment]
Hi @taquitos,

Yes, we have sent them screenshots from web browsers. We have even created a movie while we are trying to log in (they have asked for it after the first email) and as you can see this didn't help at all.

**Comment by sebastiansondergaard** [comment]
Hi.

What is the current status of this?

I started experiencing this today. I'm working in Visual Studio for Mac on a Xamarin app. Visual Studio suddenly started to report the following errors even though the account already existed:
"Error adding account - <account>
Authentication service is unavailable."

After a lot of debugging and trying removing and adding certificates I remembered that Visual Studio is using Fastlane to download certificates etc. I was glad to find this post and tried the solution of changing my IP using VPN which worked for me.

Are there anything that I can do to help you fix this error?

**Comment by taquitos** [comment]
@sebastiansondergaard the current state is still: https://github.com/fastlane/fastlane/issues/10793#issuecomment-342999930
Although, at this point we've pivoted to working on: https://github.com/fastlane/fastlane/issues/10930 because tester management and build trains are totally broken due to API changes.

**Comment by jannehansen** [comment]
Hi all,

just confirming that the workaround documented previously:

- Login using a different account from same IP
- Change IP address (VPN/Proxy)
- Login from a different IP, it unblocks your original IP.
- Login in developer.apple.com and then navigate to itunesconnect.apple.com

Worked for me as well. The key seemed to be to use the VPN - didn't work without it. Downloaded Avast SecureLine VPN from the Mac App Store, and then started to work like a charm. Thanks!

**Comment by tehnoir** [comment]
As another date point in case it's helpful, I ran into this for the first time today. I have a script doing the following:

1. Login to an account
2. Add a device to an account
3. Add the device to an array of provisioning profiles.
4. Download each of those provisioning profiles.

After step 3 (which includes 6 provisioning profiles in this failing case), it fails when it tries to download the first profile in step 4, showing the 503 Temporarily Unavailable error.

**Comment by julestburt** [comment]
Having same issue here. We use match - and are seeing this same issue w/ match and spaceship, on our bitrise build server. We create white label apps, so are using match to download certs etc, then  spaceship and pilot - all of which are failing, intermittently - all around the time it's logging into iTunes. Same error in build logs...<head><title>503 Service Temporarily Unavailable</title></head>.

Would love a fix for this issue. It's affecting all our building presently. Currently investigating setting up a vpn to mask our IP - read that this may help?? Would that work?

**Comment by julestburt** [comment]
Thanks @tehnoir. For our white-label app building on bitrise - I added a login with a dummy developer from our account, before proceeding with each build.

So no VPN needed as having another user login before the regular one that does the build (using fastlane) seemed to work.

Thanks...Jules.

**Comment by julestburt** [comment]
And today - the losing with a dummy developer account - doesn't help fix the 503 issue. I'm seeing this failure on all our bitrise builds today. :-(

**Comment by KrauseFx** [comment]
Hey everyone, thanks again for looking into this issue. Sorry it took so long, I finally found some time to look further into this issue, and was able to reproduce the issue and prepare a fix. You can check out the PR over here https://github.com/fastlane/fastlane/pull/11108

I'd love to ship this as a new release tomorrow morning 👍

**Comment by ohayon** [comment]
We shipped a new release thank to @KrauseFx's PR #11108. We'd love to have you all confirm that this works so we can close this issue! 😄 🚀

**Comment by scalessec** [comment]
I can confirm that this is now fixed for me in the `2.68.2` release. I have lane that calls `match` 3 times for an app with 7 targets. Previously this would fail, but all good now! 🎉🍺

**Comment by klaspetersson** [comment]
We are also still getting this error on fastlane 2.68.2 after some actions:

```+------+---------------------+-------------+
|             fastlane summary             |
+------+---------------------+-------------+
| Step | Action              | Time (in s) |
+------+---------------------+-------------+
| 1    | Verifying required  | 0           |
|      | fastlane version    |             |
| 2    | default_platform    | 0           |
| 3    | register_devices    | 3           |
| 4    | register_devices    | 3           |
| 5    | register_devices    | 3           |
| 6    | register_devices    | 2           |
| 7    | register_devices    | 3           |
| 8    | register_devices    | 2           |
| 9    | register_devices    | 2           |
| 💥   | register_devices    | 0           |
| 💥   | match               | 9           |
| 💥   | match               | 6           |
| 💥   | match               | 7           |
| 💥   | match               | 7           |
| 💥   | match               | 7           |
| 💥   | match               | 7           |
| 💥   | match               | 9           |
| 💥   | match               | 6           |
| 💥   | match               | 8           |
| 💥   | match               | 9           |
| 💥   | match               | 10          |
| 💥   | match               | 7           |
| 💥   | match               | 9           |
| 💥   | match               | 9           |
| 💥   | match               | 9           |
| 💥   | match               | 9           |
| 💥   | match               | 9           |
| 💥   | match               | 12          |
| 💥   | match               | 9           |
| 💥   | match               | 9           |
| 💥   | match               | 6           |
| 💥   | match               | 7           |
| 💥   | match               | 9           |
| 💥   | match               | 6           |
| 💥   | match               | 6           |
| 💥   | match               | 9           |
| 💥   | match               | 9           |
| 38   | match               | 7           |
+------+---------------------+-------------+
```
And the error is the same as usual:
```
<html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body bgcolor="white">
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>Shield</center>
</body>
</html>
```

**Comment by taquitos** [comment]
Hey @klaspetersson thanks for reporting this. It would help to submit new issues instead of commenting, that way we can collect some more information about what is going on in your specific case since the fix we shipped should take care of most situations people encounter.

**Comment by julianvogels** [comment]
We just nuked both development and distribution, and running a series of 7 match commands for different certificates and provisioning profiles returned the usual 503 error. 

Then I changed the match call to use an array of app identifiers and reduced the invocations of match from 7 to 3. Now it works.

From the match documentation:
`match(app_identifier: ["tools.fastlane.app", "tools.fastlane.app.today_widget"], type: "appstore")`

**Comment by elorz007** [comment]
In any case, more aggressive caching will not solve all the issues. For example, we have something like 8 CI nodes across different machines. Caching might ease the 503 error on one machine but between multiple machines sharing one account it won't do much.

We could have one account for each node but we would rather not...

**Comment by taquitos** [comment]
@elorz007 yeah, that stinks. I wonder, maybe there is some sort of solution where you could share sessions across all the nodes? fastlane writes session data to a file, so maybe there is a way to login on one node, copy the session to all the other nodes, and then proceed?

**Comment by casz** [comment]
where is the file written @taquitos could you perhaps share it between the same Mac user because it's in the user folder and you can just map it to a network shared folder? So they all have the same read/write access :)

**Comment by taquitos** [comment]
So all that stuff happens in: https://github.com/fastlane/fastlane/blob/bc4105dbe77d244e7d6c3ba42af2d5f13146e4da/spaceship/lib/spaceship/client.rb

`fetch_olympus_session`, `load_session_from_env` are the methods involved

also: https://github.com/fastlane/fastlane/blob/bc4105dbe77d244e7d6c3ba42af2d5f13146e4da/spaceship/lib/spaceship/two_step_client.rb#L85

where it creates a temp file:
`file = Tempfile.new('cookie.yml')`

So I think if we made this less temporary, and provided a path, a shared storage area would be good, as long as you control for multiple readers/writers to the file.

**Comment by casz** [comment]
That could be great 👍 Our Mac fleet would be appreciative

**Comment by yfedin** [comment]
Just an FYI - the `SPACESHIP_COOKIE_PATH` solution looks to be working. At least in my case. Thanks!

**Comment by casz** [comment]
@KrauseFx xcode-install does not expect the `SPACESHIP_COOKIE_PATH` 👎 😭

**Comment by casz** [comment]
I had `SPACESHIP_COOKIE_PATH=/Volumes/Fastlane fastlane deliver download_metadata` generate the cookie so I could try and unleash the `xcversion update; xcversion install 9.2` on our fleet of mac minis so we could get them updated. Worked like a charm 👍

**Comment by cerupcat** [comment]
A single `match` call with 7 identifiers is failing for me with `503`. What're the current recommendations for how to get past this? Should I split my match call into multiple `match` calls and wait X seconds between them?

**Comment by imanf** [comment]
I'm also running into this issue after upgrading Fastlane to 2.70.2, when trying to run an action across multiple teams. Sample code snippet is below. Essentially, I have several teams/accounts that are linked together, and I have a properties file where team details are stored, so `ENV['FASTLANE_USER']` will always be the same. When I run my lane, after the loop iterates through a couple of the teams, I get the 503 for subsequent iterations. 

```
before_each do |lane, options|
	if !$team.to_s.empty?
		ENV['FASTLANE_TEAM_ID'] = AppProperties::Properties[$team][:TeamID]
		ENV['FASTLANE_USER'] = AppProperties::Properties[$team][:Username]
	end
end


lane :create_app_id do
	teams = 'team1 team2 team3 team4'.split(' ')

	teams.each do |team_to_run|
		$team = team_to_run
		produce(
			app_identifier: "#{$team_to_run}.extension",
			app_name: "#{$team_to_run} Extension",
		}
	end
end
```

**Comment by casz** [comment]
We have a similar setup to @imanf :) Just a lot bigger.

All of our teams with all our app identifiers in an array and we run that each time anyone adds a new bundle id 😢
In this Fastlane repo, we also have all our devices. 
Using all of the information together with produce, match, register_devices depending on the state 💜 

```ruby
module AppsConstant
  TYPES = %w[development adhoc appstore enterprise].freeze

  TEAMS = {
    widex: {
      name: 'Widex A/S',
      git_branch: 'widex',
      supported_types: %w[development adhoc appstore].freeze,
      app_identifier: [
        'com.widex.app1',
        'com.widex.app1.watchkitapp',
        'com.widex.app1.watchkitapp.watchkitextension',
        'com.widex.app2',
        'com.widex.app3',
      ],
    },
    widexenterprise: {
      name: 'Widex Holding A/S',
      git_branch: 'widexenterprise',
      enterprise: true,
      supported_types: %w[development enterprise].freeze,
      app_identifier: [
        'com.widex.enterprise.appX',
      ],
    },
    testcompany: {
      name: 'Widex A/S',
      git_branch: 'widextesters',
      skip: true,
      supported_types: %w[development].freeze,
      app_identifier: [
        'com.widex.*',
      ],
    },
  }.freeze
end
```

**Comment by sungjin0213** [comment]
This is still occurring for me. When I try to use `fastlane pem` multiple times in a row, this is happening.

**Comment by imanf** [comment]
Are there any updates on this? I'm including another example where this is happening. I'm creating 10 provisioning profiles using sigh. The first 9 were created successfully, but the 10th one failed with a 503 error. 

```
lane :create_provisioning_profiles do
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}", provisioning_name: "#{$app} App Store", filename: "#{$app}_App_Store.mobileprovision", output_path: "#{$output_path}/profiles/")
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}", provisioning_name: "#{$app} Ad Hoc", filename: "#{$app}_Ad_Hoc.mobileprovision", output_path: "#{$output_path}/profiles/", adhoc: true)
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.TodayView", provisioning_name: "#{$app} TodayView App Store", filename: "#{$app}_TodayView_App_Store.mobileprovision", output_path: "#{$output_path}/profiles/")
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.TodayView", provisioning_name: "#{$app} TodayView Ad Hoc", filename: "#{$app}_TodayView_Ad_Hoc.mobileprovision", output_path: "#{$output_path}/profiles/", adhoc: true)
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.watchkitapp", provisioning_name: "#{$app} WatchKitApp App Store", filename: "#{$app}_WatchKitApp_App_Store.mobileprovision", output_path: "#{$output_path}/profiles/")
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.watchkitapp", provisioning_name: "#{$app} WatchKitApp Ad Hoc", filename: "#{$app}_WatchKitApp_Ad_Hoc.mobileprovision", output_path: "#{$output_path}/profiles/", adhoc: true)
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.watchkitapp.extension", provisioning_name: "#{$app} WatchKitApp Extension App Store", filename: "#{$app}_WatchKitApp_Extension_App_Store.mobileprovision", output_path: "#{$output_path}/profiles/")
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.watchkitapp.extension", provisioning_name: "#{$app} WatchKitApp Extension Ad Hoc", filename: "#{$app}_WatchKitApp_Extension_Ad_Hoc.mobileprovision", output_path: "#{$output_path}/profiles/", adhoc: true)
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.notificationserviceextension", provisioning_name: "#{$app} NotificationServiceExtension App Store", filename: "#{$app}_NotificationServiceExtension_App_Store.mobileprovision", output_path: "#{$output_path}/profiles/")
	sigh(skip_install: true, force: true, app_identifier: "#{$app_identifier}.notificationserviceextension", provisioning_name: "#{$app} NotificationServiceExtension Ad Hoc", filename: "#{$app}_NotificationServiceExtension_Ad_Hoc.mobileprovision", output_path: "#{$output_path}/profiles/", adhoc: true)
end
```

**Comment by KrauseFx** [comment]
Hey everyone, sorry we're not actively working on this right now. We've reported the issue with Apple, but haven't heard back. 

For now as a temporary workaround, can you use `sleep(10)` in-between calls to make it work? Or is that not enough?

**Comment by KrauseFx** [comment]
It does seem like many people are affected by this... can you confirm the above workaround doesn't work, and I'll see if I can find a better solution for this.

**Comment by imanf** [comment]
Yes, that workaround does work and would be fine for a temporary
workaround. Thanks!

**Comment by imodeveloper** [comment]
@KrauseFx I have similar problem with deliver.

```
+-----------------------+-----------------------+
| Detected Values from './fastlane/Deliverfile' |
+-----------------------+-----------------------+
| app_identifier        | my.app.id           |
| username              | my@email.com |
+-----------------------+-----------------------+

[19:42:19]: Login to iTunes Connect (my@email.com)
+------------------+--------------------------+
|                [33mLane Context[0m                 |
+------------------+--------------------------+
| DEFAULT_PLATFORM | ios                      |
| PLATFORM_NAME    | ios                      |
| LANE_NAME        | ios deliver_to_app_store |
| BUILD_NUMBER     | 122                      |
+------------------+--------------------------+
[19:42:22]: [31m<html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body bgcolor="white">
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>Shield</center>
</body>
</html>
```

**Comment by daniel-beard** [comment]
@KrauseFx that workaround does not work for me when using more than a single account. Seems to still be an issue with multiple accounts :(

**Comment by scorpion35** [comment]
@KrauseFx My account just got locked too :( 

I am using fastlane version `2.75.1`, and tried to use `Spaceship::Tunes.login("my apple id")` and received `503 Service Temporarily Unavailable` error. 

Please help.

**Comment by slhodson969** [comment]
@scorpion35 For next time maybe, we found that if we turned on our VPN we could get around the issue....until it gets blocked too.

We have now added `sleep(15)` in between our multiple `pem` calls also to try and avoid this happening at all and so far it seems to have worked.

**Comment by scorpion35** [comment]
@slhodson969 Yea tried VPN option as well yesterday, didn't seem to work. That's the reason I had to go with changing my password and updated Fastlane to `2.75.1`. All is fine now! 

Thank you, everyone, for amazing work on Fastlane/Spaceship and bug fixes!!

**Comment by sethsamuel** [comment]
We're hitting this pretty constantly to the point where we can't get fastlane setup. Changing password/toggling VPN sometimes fixes it, but that's obviously impractical to do every several runs when trying to debug new settings.

**Comment by daniel-beard** [comment]
Issue is not resolved.

**Comment by taquitos** [comment]
Update: I've updated the radar I filed with Apple last year around this (https://openradar.appspot.com/35445457), they just asked me for more information and to see if I can repro (which I can). So I updated them. 🤞 they fix it.

**Comment by scalessec** [comment]
Chill @fastlane-bot, it's still not resolved.

**Comment by elorz007** [comment]
I talked to the Apple engineers today at WWDC to try to get some answer to this problem. Because I suspect that even if fastlane moves to the new API, we will still get problems with one Apple ID logging in many times in a some minutes or logging in from different machines.

They told me that because Fastlane is a third party tool, they can only look into the issue if the login problems also occur on some of their tools (e.g. Application Loader, Xcode, ...). Because being able to log in with one account multiple times or across many devices is something they really want to avoid for obvious security reasons.

So I will try to reproduce this by logging in and out of Application Loader. If I reproduce the issue they will take a look.

**Comment by elorz007** [comment]
According to the new WWDC session ["Automating App Store Connect"](https://developer.apple.com/videos/play/wwdc2018/303/) the authentication to the API will be done like this:
1. Create API key pair. Public key will be stored on Apple. Keys will be created in the itunes connect website. Private keys are generated by apple (wtf!) and you will be able to download them from the page once.
2. Generate temporary tokens with the id of the private key and the issuer id. (should be easy with JWT)
3. Send token with the requests inside the auth header.

It will be available this summer.

**Comment by scalessec** [comment]
It’s still broken, @fastlanebot

**Comment by yfedin** [comment]
@scalessec, Are you using SPACESHIP_COOKIE_PATH to cache your session? I've been running into this issue all the time up till latest 2.99.0 release where session caching finally started working. In my multi-node build environment - I have set up a shared folder to store cached session cookies. Then - my Jenkins build script for most of the jobs basically looks like this:

`export PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
export LANG=en_US.UTF-8
export FASTLANE_DISABLE_COLORS=1
export SPACESHIP_COOKIE_PATH=$JENKINS_HOME/cookies

security unlock-keychain -p $KC_PWD $KC_PATH

bundle install --no-color --path .bundle
bundle exec --no-color fastlane spaceauth
bundle exec --no-color fastlane ci_build`

**Comment by daniel-beard** [comment]
Calm it @fastlane-bot, this is still an issue.

**Comment by xhairbait** [comment]
I think the issue has been resolved today.

I ran fastlane update_fastlane at 10:50 AM and then did a "fastlane deliver init" and it connected.

**Comment by daniel-beard** [comment]
@xhairbait are you using a single Apple ID across multiple accounts? That's where this issue is causing the most problems for me.

**Comment by scalessec** [comment]
Still an issue @fastlane-bot

**Comment by daniel-beard** [comment]
Still an issue.

**Comment by scalessec** [comment]
Still broken.

**Comment by scalessec** [comment]
Bump

**Comment by daniel-beard** [comment]
Still an issue, go away fastlane bot

**Comment by max-ott** [comment]
Hi @daniel-beard we had this issue previously, but it's no longer happening. Could you let us know your current / latest `fastlane` setup and when does the error occur for you?

**Comment by daniel-beard** [comment]
Hey @max-ott I haven't yet upgraded to the version that moved a lot of the calls to app store connect. I'll plan to update next weekend and re-test.
